py/runtime_utils.c

```diff
@@ -27,7 +27,7 @@
 
 #include "py/runtime.h"
 
-void mp_call_function_1_protected(mp_obj_t fun, mp_obj_t arg) {
+void __attribute__((noinline)) mp_call_function_1_protected(mp_obj_t fun, mp_obj_t arg) {
```

Comment:
See how this is done in `py/stackctrl.h`. We use `__asm volatile ("");`, which is more of a guarantee than `__attribute__((noinline))` (which is ignored in certain circumstances.). Also, if you want to declare as noinline, the macro `MP_NOINLINE` is available, and is cross-platform.